lib/rubocop/cop/style/multiple_comparison.rb

```diff
@@ -0,0 +1,80 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks against comparing a variable with multiple items,
+      # where `Array.include?` could be used instead to avoid DRY errors.
+      #
+      # @example
+      #   # bad
+      #   if a == 'a' || a == 'b' || a == 'c'
+      #
+      #   # good
+      #   if ['a', 'b', 'c'].include? a
+      class MultipleComparison < Cop
+        MSG = 'Avoid comparing a variable with multiple items' \
+          'in a conditional, use include? instead.'.freeze
+
+        def on_if(node)
+          return unless nested_variable_comparison?(node.condition)
+          add_offense(node, :expression)
+        end
+
+        private
+
+        def_node_matcher :simple_double_comparison?, '(send $lvar :== $lvar)'
+        def_node_matcher :simple_lhs_comparison?, '(send $lvar :== _)'
+        def_node_matcher :simple_rhs_comparison?, '(send _ :== $lvar)'
+
+        def nested_variable_comparison?(node)
+          return false unless nested_comparison?(node)
+          variables_in_node(node).count == 1
+        end
+
+        def variables_in_node(node)
+          if node.or_type?
+            parts = node.node_parts
```

Comment:
I don't think this temporary variable adds to readability. Also `#map` + `#flatten` can be `#flat_map`. 😊 

```
node.node_parts.flat_map { |p| variables_in_node(part) }.uniq
```